### PR TITLE
Chore/release process improvements

### PR DESCRIPTION
**Pull request by nellyk**
## Description

- Fix the current issue with the docker release script
- Use [conventational-changelog](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-cli) to generate new changelogs as `standard version` has an open issue https://github.com/conventional-changelog/standard-version/issues/509, which generates empty changelogs.

## Motivation and Context

Improve the One App release process

## How Has This Been Tested?

- with changelog changes --> https://github.com/nellyk/one-app/pull/92 

- docker release run https://github.com/nellyk/one-app/actions/runs/95351257

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (adding or updating documentation)

## Checklist:

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

No impact, improves the release process for the One App

### CODE REVIEW

File: .github/workflows/pull_request_release.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - run: git fetch --prune --unshallow
```
Comment (nellyk) [inline .github/workflows/pull_request_release.yml:12]: This provides the action with access to all the tags and git history, allowing changelog to be generated with all changes going into the release

File: package.json [hunk 2]
```diff
@@ -180,7 +182,8 @@
       "one-app-statics/package.json"
     ],
     "skip": {
-      "tag": true
+      "tag": true,
+      "changelog": true
```
Comment (nellyk) [inline package.json:186]: Skipping the changelog generation as we do not need standard version to generate this

File: .github/workflows/pull_request_release.yml [hunk 3]
```diff
@@ -24,7 +25,10 @@ jobs:
           NODE_ENV=development npm ci
           git config --local user.email "one.amex@aexp.com"
           git config --local user.name "OneAmexBot"
-          npm run release
+          npm run release -- --no-verify
```
Comment (infoxicator) [inline .github/workflows/pull_request_release.yml:28]: what does the `--no-verify` flag do?

File: RELEASE.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ Releases can be triggered in two ways:
...
+ 1. An automated pull request will be raised every Wednesday at 16:00 UTC, from a `prepare-release` branch to master. This uses [pull_request_release workflow](.github/workflows/pull_request_release.yml) and updates the [package.json](package.json), [package-lock.json](package-lock.json), [one-app-statics package.json](one-app-statics/package.json) and runs `npm run release:changelog` to generate the changelog. Behind the scene it uses [standard-version](https://github.com/conventional-changelog/standard-version) to generate and update these files with the changes for released. The commit message contains will be in the following format `chore(release): X.X.X` . You can update this pull request to remove or add any new changes.
```
Comment (infoxicator) [inline RELEASE.md:12]:
```suggestion
 1. An automated pull request will be raised every Wednesday at 16:00 UTC, from a `prepare-release` branch to master. This uses [pull_request_release workflow](.github/workflows/pull_request_release.yml) and updates the [package.json](package.json), [package-lock.json](package-lock.json), [one-app-statics package.json](one-app-statics/package.json) and runs `npm run release:changelog` to generate the changelog. Behind the scenes it uses [standard-version](https://github.com/conventional-changelog/standard-version) to generate and update these files with the changes to be released. The commit message will be in the following format `chore(release): X.X.X` . You can update this pull request to add or remove any new changes.
```

File: .github/workflows/pull_request_release.yml (hunk 3 above)
Comment (nellyk) [reply .github/workflows/pull_request_release.yml:28]: This would skip the hooks as we don't need this in this case

File: RELEASE.md [hunk 5]
```diff
@@ -25,13 +25,33 @@ Releases can be triggered in two ways:
 2. After the the generated tag is pushed to the branch this will trigger the docker build and publish the statics and push the images to Docker Hub. The development and production images would be accessible in docker [https://hub.docker.com/u/oneamex](https://hub.docker.com/u/oneamex)
 
 3. We are currently using [https://github.com/release-drafter/release-drafter](https://github.com/release-drafter/release-drafter) to generate release notes. Please add the labels specified within [release-drafter](.github/release-drafter.yml) to categorize different pull requests. Update the draft release notes and tie it to the release tag above, you can also link this to different artifacts. The statics assets will be published and added to a tag that has been released.
+4. For the release notes use the below format, checkout [https://github.com/americanexpress/one-app/releases](https://github.com/americanexpress/one-app/releases)
+   
+   ```
+
```
Comment (JAdshead) [inline RELEASE.md:31]: remove this whitespace

File: RELEASE.md [hunk 6]
```diff
@@ -25,13 +25,33 @@ Releases can be triggered in two ways:
 2. After the the generated tag is pushed to the branch this will trigger the docker build and publish the statics and push the images to Docker Hub. The development and production images would be accessible in docker [https://hub.docker.com/u/oneamex](https://hub.docker.com/u/oneamex)
 
 3. We are currently using [https://github.com/release-drafter/release-drafter](https://github.com/release-drafter/release-drafter) to generate release notes. Please add the labels specified within [release-drafter](.github/release-drafter.yml) to categorize different pull requests. Update the draft release notes and tie it to the release tag above, you can also link this to different artifacts. The statics assets will be published and added to a tag that has been released.
+4. For the release notes use the below format, checkout [https://github.com/americanexpress/one-app/releases](https://github.com/americanexpress/one-app/releases)
```
Comment (JAdshead) [inline RELEASE.md:28]: what do you mean by "checkout" ?
It's not very clear what is meant to happen with the release notes. where do they come from ? how do you create them ? what is the complete process

File: package.json [hunk 7]
```diff
@@ -36,6 +36,7 @@
     "set-middleware": "set-middleware",
     "set-dev-endpoints": "node scripts/set-dev-endpoints.js",
     "postinstall": "npm run build",
+    "release:changelog": "echo -e \"$(head -3 CHANGELOG.md)\n\n$(node_modules/.bin/conventional-changelog -p angular)\n\n$(tail -n +3 CHANGELOG.md)\" > CHANGELOG.md",
     "release": "standard-version"
```
Comment (JAdshead) [inline package.json:40]: this should probably be
`"release:standard-version": "standard-version -- --no-verify"`
then have release call both
"release": "release:standard-version && release:changelog"

File: .github/workflows/pull_request_release.yml (hunk 3 above)
Comment (JAdshead) [reply .github/workflows/pull_request_release.yml:28]: see my below comment in package.json

File: RELEASE.md (hunk 6 above)
Comment (nellyk) [reply RELEASE.md]: Updated this to be more clear, let me know what you think. Provided a template for the release notes below this.

File: package.json (hunk 7 above)
Comment (nellyk) [reply package.json]: Thank you, yes this works! updated

File: RELEASE.md (hunk 5 above)
Comment (nellyk) [reply RELEASE.md]: Updated 0415f55

**Comment by nellyk** [comment]
> Really digging the documentation @nellyk.

Thank you 😄

File: RELEASE.md [hunk 8]
```diff
@@ -21,17 +21,37 @@ Releases can be triggered in two ways:
...
 
 ## Manual release process
 
 This process can be used to make ad hoc releases outside of wednesday release cycle.
 
- 1. Run `npm run release` locally within your branch, this would update the changelog, [package.json](package.json), [package-lock.json](package-lock.json)and [one-app-statics package.json](one-app-statics/package.json) with the new version to be released. Push your changes and create a pull request to master.
- 2. When the changes are merged and reviewed. The same process from step 3 above will be followed.
+ 1. Run `npm run release` locally within your branch, this would update [package.json](package.json), [package-lock.json](package-lock.json)and [one-app-statics package.json](one-app-statics/package.json) with the new version to be released. 
+ 2. Run `npm run release:changelog` to generate and update the changelog.
+ 3. Push your changes and create a pull request to master.
```
Comment (Francois-Esquire) [inline RELEASE.md:53]:
```suggestion
 3. Commit using the `chore(release): X.X.X` message format and push your changes
 and create a pull request to master.
```

File: RELEASE.md [hunk 9]
```diff
@@ -41,16 +61,16 @@ If changes were made and need to be reverted. Please use the [manual release pro
 
 ### How can I run the first release?
 
-For the first release please use the [manual release process](#manual-release-process). Run `npm run release -- --first-release` to generate the initial changelog and update the package.json files.
+For the first release please use the [manual release process](#manual-release-process). Run `npm run release -- --first-release` to update the package.json files.
 
 ### How can I create a prerelease?
 
-For the first release please use the [manual release process](#manual-release-process). Run `npm run release -- --prerelease` to generate the initial changelog and update the package.json files.
+For the first release please use the [manual release process](#manual-release-process). Run `npm run release -- --prerelease` to update the package.json files.
```
Comment (Francois-Esquire) [inline RELEASE.md:69]:
```suggestion

For dry running the output from release, you can run `npm run release -- --prerelease --dry-run`.
```

File: RELEASE.md [hunk 10]
```diff
@@ -41,16 +61,16 @@ If changes were made and need to be reverted. Please use the [manual release pro
...
```
Comment (Francois-Esquire) [inline RELEASE.md:68]:
```suggestion
For the first release please use the [manual release process](#manual-release-process). Run `npm run release -- --prerelease` to update the package.json files.

For dry running the output from release, you can run `npm run release -- --prerelease --dry-run`.
```

File: RELEASE.md (hunk 8 above)
Comment (Francois-Esquire) [reply RELEASE.md:53]: Would this need to be on a specific branch?

Comment (nellyk) [reply RELEASE.md:53]: nope, although for consistency using the `prepare-release` branch would be a better suggestions. I'll mention that

File: RELEASE.md [hunk 11]
```diff
@@ -21,28 +21,28 @@ Releases can be triggered in two ways:
    > Integration tests will continue to run in Travis as they currently do, after the automated pull request is created it would run the tests on `prepare-release` branch.
 
 1. The merge will trigger the automatic generation of a new tag using the semantic version provided during the merging of the pull request above.
-
 2. After the the generated tag is pushed to the branch this will trigger the docker build and publish the statics and push the images to Docker Hub. The development and production images would be accessible in docker [https://hub.docker.com/u/oneamex](https://hub.docker.com/u/oneamex)
+3. The statics assets will be published and added to a tag that has been released via the [release static assets action](.github/workflows/docker_release.yml)
```
Comment (JAdshead) [inline RELEASE.md:25]:
```suggestion
3. The statics assets will be published and added to the tag created via the [release static assets action](.github/workflows/docker_release.yml) as part of the previous step.
```

File: RELEASE.md [hunk 12]
```diff
@@ -21,28 +21,28 @@ Releases can be triggered in two ways:
...
+4. We are currently using [https://github.com/release-drafter/release-drafter](https://github.com/release-drafter/release-drafter) to generate release notes. This is a github action that generates draft releases note that can be added to the released tag. Please add the labels specified within [release-drafter](.github/release-drafter.yml) to categorize different pull requests to ensure that any changes are categorized correctly. You can also use the changelog generated for this content.
```
Comment (JAdshead) [inline RELEASE.md:26]: does release-drafter have to use labels or can it use commits ? can we auto label based of the commit.
If we take the changes add as part of the changelog do we even need release drafter ?

File: RELEASE.md [hunk 13]
```diff
@@ -21,28 +21,28 @@ Releases can be triggered in two ways:
...
+5. Update the draft release notes with any changes required and tie it to the released tag.
```
Comment (JAdshead) [inline RELEASE.md:27]: what do you mean by "tie it" ? 
```suggestion
5. Update the draft release notes with any changes required and edit the released tag to add them there.
```

File: RELEASE.md [hunk 14]
```diff
@@ -21,28 +21,28 @@ Releases can be triggered in two ways:
...
-3. We are currently using [https://github.com/release-drafter/release-drafter](https://github.com/release-drafter/release-drafter) to generate release notes. Please add the labels specified within [release-drafter](.github/release-drafter.yml) to categorize different pull requests. Update the draft release notes and tie it to the release tag above, you can also link this to different artifacts. The statics assets will be published and added to a tag that has been released.
-4. For the release notes use the below format, checkout [https://github.com/americanexpress/one-app/releases](https://github.com/americanexpress/one-app/releases)
-   
-   ```
-
-    [Paste changelog entries here]
-  
-    ### Docker Images
+  ```
+      [Paste changelog entries here]
```
Comment (JAdshead) [inline RELEASE.md:31]: is all this white space necessary ?

Comment (nellyk) [reply RELEASE.md:31]: Remove this, i had missed this earlier

File: RELEASE.md (hunk 12 above)
Comment (nellyk) [reply RELEASE.md]: We have some work to address this in a separate PR

**Pull request merged**